src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/BoxingHelpers.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Runtime.CompilerServices
+{
+    [StackTraceHidden]
+    [DebuggerStepThrough]
+    internal static unsafe partial class BoxingHelpers
```

Comment:
Most of these types are going to be loaded on startup path. It feels a bit too fine grained to have separate type for the few unboxing helpers. 

Would it be better for the unboxing helpers to live in CastHelpers? They are coupled anyway.

Reply:
Sure, seems reasonable to me. I had thought I might be moving the allocation based helpers over too so this class wouldn't be so small, but honestly, just copying the assembly routines from NativeAOT seems like a better path for most of those.